sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/UcfgInstructionTest.cs

```diff
@@ -862,5 +862,37 @@ public override bool Equals(object obj)
 }";
             UcfgVerifier.VerifyInstructions(code, "Equals");
         }
+
+        [TestMethod]  // Regression test for https://jira.sonarsource.com/browse/SONARSEC-199
+        public void Peach_Exception_UnexpectedTypeOfTargetForTheInstruction_Akka()
+        {
+            // Adpated from akka\src\core\Akka\IO\SocketEventArgsPool.cs
+            const string code = @"
+namespace Akka.IO
+{
+    using System;
+    using System.Net.Sockets;
+     internal class PreallocatedSocketEventAgrsPool
+    {
+        private EventHandler<SocketAsyncEventArgs> _onComplete = null;
+         private SocketAsyncEventArgs CreateSocketAsyncEventArgs()
+        {
+            var e = new SocketAsyncEventArgs { UserToken = null };
+                // %0 := new System.Net.Sockets.SocketAsyncEventArgs
+                // %1 := System.Net.Sockets.SocketAsyncEventArgs.SocketAsyncEventArgs() [ %0 ]
+                // %2 := System.Net.Sockets.SocketAsyncEventArgs.UserToken.set [ %0 const ]
+                // e := __id [ %0 ]
+
+            e.Completed += _onComplete;
+                // %3 := __id [ this._onComplete ]
+                // %4 := System.Net.Sockets.SocketAsyncEventArgs.Completed.add [ System.Net.Sockets.SocketAsyncEventArgs const %3 ]
```

Comment:
It seems a bit odd that there isn't a reference to _e_ in %4. Still, I guess there's no way adding the event handler could taint _e_ (i.e. if _this_ was tainted).

Reply:
I actually didn't noticed that `e` was missing. It's more corner case so I will ignore it for now but create a ticket about it.